File: Strings/notes.py
#Lisa Rehm, Strings Notes Python

# words in programming (data type) "" ''

# print("Hello world")

# print('Hello world') #you can use either "" or '' but don't mix them

#print('This is someone's code)
# print("this is someone's code")

# name = input("What is your first name?\n").strip().capitalize()
# stupid proofing is asking more straighforward questions and the sanitization
# print(f"Hello {name} welcome to my program.")

# num = input("Please give me a number with no decimal\n").strip()

# print(num + 1) 
# this doen't work because inputs are automatically strings so you have to tell it it's an integer

# num = input("Please give me a number with no decimal\n").strip()

# print(num + str(1)) 
# this doesn't work because it puts the 1 right after the input, it is because of the parentheses around the 1
# input = 5, output = 51, this is concatenation. when you add strings it put them next to each other. The fix is to convernt them to integers.

# sentence = "The quick brown fox jumps over the lazy dog."

# print(sentence.find("fox")) or if you want the whole word:

# word = sentence.find("fox)")
# print(sentence(word:word+3)) length of word
# print(sentence(4,9))
# print(len(sentence))
name = "Lisa"
percent = 89
print("Welcome to my program" +name)
print(f"Welcome to my program {name}!")
print(f"Welcome to my program {percent:.1f}!")

File: Strings/notes.c
//Lisa Rehm, Strings Notes C
#include <stdio.h>
#include <string.h>

char name[20]; //have a number if you are not introducing the value rn

int main(void){
    //printf("Please tell me your full name.\n");
    //scanf("%s", name); 
    //scanf will only take input up to the first space, then it stops
    //fgets(name, 20, stdin); //fgets is also an input but gets everything, you need to tell it 3 things: variable, char # and stdin
    //printf("Hello %sWelcome to my program!", name);
    //char sentence[] = "The quick brown fox jumps over the lazy dog";
    //printf("%s\n", sentence);
    //printf("%lu\n", sizeof(sentence)); //index number (starting at 0)
    //printf("%lu\n", strlen(sentence)); //number of actual characters
    char one[] = "Hello ";
    char two[] = "World!";
    char three[] = "This is my program. ";
    //two[5] = '?';
    printf("%s\n", one);
    strcat(one, two);
    printf("%s\n", one);
    //strcat(three, one); //when cat, make sure to not run out of characters on the variable
    //printf("%s", three);
    strcat(one, three);
    printf("%s", one);
    return 0;
}
